births-by-day-of-week.ipynb

```diff
@@ -0,0 +1,277 @@
+    "import matplotlib.pyplot as plt\n",
+    "plt.style.use('seaborn-whitegrid')\n",
+    "\n",
+    "url = \"https://raw.githubusercontent.com/jakevdp/data-CDCbirths/master/births.csv\"\n",
+    "data = pd.read_csv(url)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We need only years from 1960 to 1989\n",
+    "data = data[data[\"year\"] < 1990]\n",
+    "data = data[data[\"year\"] > 1959]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data.dropna(inplace = True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#data[\"day\"].unique()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Removing impossible dates such as day 99 of a month\n",
+    "data = data[data[\"day\"] <= 31]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#data[\"day\"].unique()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#print(pd.cut(data[\"births\"], [0, 100, 1000, 10000]).unique())\n",
+    "#print(\"\\n\", len(data[data[\"births\"] < 200]), len(data[data[\"births\"] < 2000]))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Removing all other absurd dates such as 30 of february or 31 of april, \n",
+    "# wich have a significantly lower number of birth\n",
+    "data = data[data[\"births\"] > data[\"births\"].mean()/10]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#print(len(data[data[\"births\"] < 2000]))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Transforming the 3 columns \"year\", \"month\", \"day\" into a single \"date\" one \n",
+    "data[\"date\"] = pd.to_datetime(data.loc[:, :\"day\"])\n",
+    "# We keep the year data for future grouping\n",
+    "data.drop([\"month\", \"day\"], axis = 1, inplace = True)\n",
+    "group = data.groupby([\"date\", \"year\"]).sum()\n",
+    "group[\"week_day\"] = 0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#group.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for day, year in group.index:\n",
+    "    group.loc[day, \"week_day\"] = day.weekday()"
```

Comment:
It could definitely work ! But with that solution I have to keep a "year" column as I did or else I would lose all the information concerning the decade.  
Your way does seem cleaner.